aws/resource_aws_budgets_budget.go

```diff
@@ -226,9 +331,77 @@ func resourceAwsBudgetsBudgetRead(d *schema.ResourceData, meta interface{}) erro
 
 	d.Set("time_unit", budget.TimeUnit)
 
+	return resourceAwsBudgetsBudgetNotificationRead(d, meta)
+}
+
+func resourceAwsBudgetsBudgetNotificationRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).budgetconn
+
+	accountID, budgetName, err := decodeBudgetsBudgetID(d.Id())
+
+	describeNotificationsForBudgetOutput, err := client.DescribeNotificationsForBudget(&budgets.DescribeNotificationsForBudgetInput{
+		BudgetName: aws.String(budgetName),
+		AccountId:  aws.String(accountID),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	notifications := make([]map[string]interface{}, 0)
+
+	for _, notificationOutput := range describeNotificationsForBudgetOutput.Notifications {
+		setNotificationDefaults(notificationOutput)
+		notification := make(map[string]interface{})
+		notification["comparison_operator"] = *notificationOutput.ComparisonOperator
+		notification["threshold_type"] = *notificationOutput.ThresholdType
+		notification["threshold"] = *notificationOutput.Threshold
+		notification["notification_type"] = *notificationOutput.NotificationType
+
+		subscribersOutput, err := client.DescribeSubscribersForNotification(&budgets.DescribeSubscribersForNotificationInput{
+			BudgetName:   aws.String(budgetName),
+			AccountId:    aws.String(accountID),
+			Notification: notificationOutput,
+		})
+
+		if err != nil {
+			return err
```

Comment:
We should provide error context here for operators and code maintainers:

```suggestion
			return fmt.Errorf("error describing Budget (%s) Notification Subscribers: %s", d.Id(), err)
```